generate.go

```diff
@@ -85,6 +85,19 @@ type LinkMap map[string]*Link
 // CfgMap is meant to represent a map with values of one or more unknown types
 type CfgMap map[string]interface{}
 
+func Join(cfgs ...*CfgMap) (CfgMap, error) {
+	c := CfgMap{}
+	for _, cfg := range cfgs {
+		for k, v := range *cfg {
+			if _, ok := c[k]; ok {
+				return nil, fmt.Errorf("Duplicate key found in two contexts: %s", k)
```

Comment:
I wonder if it should be an error? Might be nice to just have the last context override previous ones. Maybe a flag to change the functionality? 

Probably don't have to worry about it right now tho

Reply:
I think that might create a lot of unintended behaviour for common config types. I like the idea of a flag but I think keyname conflicts should create an error by default.